# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.h
# Commit: d3816f82c0b9
# Full Hash: d3816f82c0b9d4ceca0bb2ad177694164ed0921f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-03 21:40:10
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r 876b1031a6ae -r d3816f82c0b9 editor/libeditor/EditorBase.h
--- a/editor/libeditor/EditorBase.h	Fri Apr 03 01:38:22 2020 +0000
+++ b/editor/libeditor/EditorBase.h	Fri Apr 03 01:39:36 2020 +0000
@@ -590,7 +590,7 @@
    * All actions that have to be done when the editor is focused needs to be
    * added here.
    */
-  void OnFocus(dom::EventTarget* aFocusEventTarget);
+  MOZ_CAN_RUN_SCRIPT void OnFocus(nsINode& aFocusEventTargetNode);
 
   /** Resyncs spellchecking state (enabled/disabled).  This should be called
    * when anything that affects spellchecking state changes, such as the
@@ -604,7 +604,7 @@
    * selection state even if this has no focus.  So if destroying editor,
    * we have to call this method for focused editor to set selection state.
    */
-  void ReinitializeSelection(Element& aElement);
+  MOZ_CAN_RUN_SCRIPT void ReinitializeSelection(Element& aElement);
 
   /**
    * InsertTextAsAction() inserts aStringToInsert at selection.
@@ -2361,7 +2361,7 @@
    * XXX What's the difference with PlaceholderTransaction? Should we always
    *     use it instead?
    */
-  void BeginTransactionInternal();
+  MOZ_CAN_RUN_SCRIPT void BeginTransactionInternal();
   MOZ_CAN_RUN_SCRIPT void EndTransactionInternal();
 
  protected:  // Shouldn't be used by friend classes
@@ -2583,7 +2583,7 @@
    * a host of the editor, i.e., the editor doesn't get focus, this does
    * nothing.
    */
-  nsresult InitializeSelection(dom::EventTarget* aFocusEventTarget);
+  MOZ_CAN_RUN_SCRIPT nsresult InitializeSelection(nsINode& aFocusEventTarget);
 
   enum NotificationForEditorObservers {
     eNotifyEditorObserversOfEnd,
@@ -2638,7 +2638,7 @@
         EditorBase& aEditorBase MOZ_GUARD_OBJECT_NOTIFIER_PARAM)
         : mEditorBase(aEditorBase) {
       MOZ_GUARD_OBJECT_NOTIFIER_INIT;
-      mEditorBase.BeginTransactionInternal();
+      MOZ_KnownLive(mEditorBase).BeginTransactionInternal();
     }
 
     MOZ_CAN_RUN_SCRIPT ~AutoTransactionBatch() {